### PR TITLE
feat: add `write` pattern option (`pattern.write`)

### PR DESCRIPTION
**Pull request by evilebottnawi**

### CODE REVIEW

**Comment by evilebottnawi** [comment]
@d3viant0ne @michael-ciniawsky friendly ping

**Comment by michael-ciniawsky** [comment]
@evilebottnawi I'm not really sure about this feature tbh, it feels redundant to some point I need to think about it first

**Comment by evilebottnawi** [comment]
@michael-ciniawsky this need for server side rendering application (php, python, ruby and etc), example popular plugin for manifest have this option https://github.com/danethurber/webpack-manifest-plugin#optionswritetofileemit

File: README.md [hunk 1]
```diff
@@ -42,6 +42,7 @@ Or, in the simple case of just a `from` with the default destination, you can us
 | `ignore` | N | [] | Additional globs to ignore for this pattern |
 | `transform` | N | function(content, path) {<br>&nbsp;&nbsp;return content;<br>} | Function that modifies file contents before writing to webpack |
 | `force` | N | false | Overwrites files already in compilation.assets (usually added by other plugins) |
+| `writeWhenEmit` | N | false | If set to `true` will emit to build folder and memory in combination with `webpack-dev-server` |
```
Comment (michael-ciniawsky) [inline README.md]: Can we find a terser name for this `write || emit`?

File: src/writeFile.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import fs from 'fs';
+import fs from 'fs-extra';
```
Comment (michael-ciniawsky) [inline src/writeFile.js]: `mkdirp` instead of the 'whole' `fs-extra` dependency?

File: tests/index.js [hunk 3]
```diff
@@ -6,9 +6,13 @@ import {
 // ensure we don't mess up classic imports
 const CopyWebpackPlugin = require('./../dist/index');
 
-import fs from 'fs';
+import fs from 'fs-extra';
 import path from 'path';
 import _ from 'lodash';
+import chai from 'chai';
```
Comment (michael-ciniawsky) [inline tests/index.js]: Please no `chai` => [`jest.mock`]( http://facebook.github.io/jest/docs/en/mock-function-api.html#content) 🙃

**Comment by michael-ciniawsky** [comment]
https://www.npmjs.com/package/write-file-webpack-plugin

File: README.md (hunk 1 above)
Comment (evilebottnawi) [reply README.md]: @michael-ciniawsky yep, write seems good for me

File: src/writeFile.js (hunk 2 above)
Comment (evilebottnawi) [reply src/writeFile.js]: @michael-ciniawsky `mkdirp` have many issue with windows, `fs-extra` more stable

File: tests/index.js (hunk 3 above)
Comment (evilebottnawi) [reply tests/index.js]: @michael-ciniawsky i think we do this when merge with default

File: src/writeFile.js (hunk 2 above)
Comment (michael-ciniawsky) [reply src/writeFile.js]: kk

File: tests/index.js (hunk 3 above)
Comment (michael-ciniawsky) [reply tests/index.js]: hmm... 😞  fair point but it's still needs to be refactored again with `webpack-defaults` then 🤢. Can't be helped

**Comment by evilebottnawi** [comment]
@michael-ciniawsky seems good plugin, thanks! Let's wait other opinion about this feature

**Comment by michael-ciniawsky** [comment]
If that plugin is sufficient enough (e.g not sure if ignoring some files is possible with it) we should avoid adding more `pattern` options

**Comment by evilebottnawi** [comment]
@michael-ciniawsky i test this plugin in near future, if that plugin is sufficient enough, i will close this PR

**Comment by evilebottnawi** [comment]
Close in favor write-file-webpack-plugin

**Pull request closed without merging**